src/utils.cpp

```diff
@@ -1051,3 +1019,63 @@ QStringList Utils::get_filter_names(const QString &Source)
 	obs_source_release(source);
 	return enumParams.names;
 }
+
+/**
+ * Hotkey model
+ */
+int HotkeyModel::rowCount(const QModelIndex &parent) const
+{
+	return hotkeysList.size();
+}
+int HotkeyModel::columnCount(const QModelIndex &parent) const
+{
+	return 1;
+}
+QVariant HotkeyModel::data(const QModelIndex &index, int role) const
+{
+	if (!index.isValid())
+		return QVariant();
+
+	if (index.row() >= hotkeysList.size())
+		return QVariant();
+
+	if (role == Qt::DisplayRole) {
+		return QVariant(obs_hotkey_get_description(hotkeysList[index.row()]));
+	}
+	return QVariant();
+}
+void HotkeyModel::fetchHotkeys()
+{
+	beginResetModel();
+	hotkeysList.clear();
+
+	obs_enum_hotkeys(
+		[](void *data, obs_hotkey_id id, obs_hotkey_t *obsHotkey) {
+			if (obs_hotkey_get_registerer_type(obsHotkey) != OBS_HOTKEY_REGISTERER_FRONTEND ||
+			    QString(obs_hotkey_get_name(obsHotkey)).contains("OBSBasic"))
+				return true;
+			auto hotkeysList = static_cast<QList<obs_hotkey_t *> *>(data);
+			hotkeysList->append(obsHotkey);
+			return true;
+		},
+		&hotkeysList);
+
+	endResetModel();
+}
+obs_hotkey_t *HotkeyModel::getHotkeyAtIndex(int index)
```

Comment:
not needed for the qcombobox, just use QComboBox's currentData

Reply:
Ok interesting, gonna try it out

Reply:
It didn't work. `CurrentData` gives only data back, if you store it with `addItem()` in second parameter. We don't use addItem at all, we use models instead. And I didn't found another way to get selected item